### PR TITLE
Add documentation for Custom/SQL metric templates.

### PR DESCRIPTION
**Pull request by janet-can**
The root information for this document is in [Slite](https://sodadata.slite.com/app/channels/3F5HDIlx4V/notes/ixf_W6jAz).

To review:
- That the "Explain the SQL" sections are accurate.
- That the templates use `type: failed_row` when appropriate for the use case.
- That the structure (order of content under each H2) is consistently repeated throughout document.
- That the overall flow of the document is reasonable and useful.

### CODE REVIEW

**Review by bastienboutonnet — COMMENTED** [review]
This is generally pretty good. I added a few questions as comment and will be making a PR against this branch with a few misc fixes and "FIXME/TODO" messages for us to check before merging.

File: soda-sql/custom-metric-templates.md [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+
+## Find and display duplicate values
+
+Though you can use the built-in metric `duplicate_count` to test the contents of a column of unique IDs for duplicate values, a test passes or fails according to the number of duplicate values that exist. In other words, the metric reveals how many rows have duplicate content, but not which rows. (See an [example]({% link soda-sql/examples-by-metric.md %}#duplicate_count) of a test that uses `duplicate_count`.)
+
+If you use Soda Cloud and have [connected it]({% link soda-cloud/connect_to_cloud.md %}) to your instance of Soda SQL, you can use a custom metric to explicitly send failed rows that contain duplicate IDs to Soda Cloud. (Learn how to [examine failed rows]({% link soda-cloud/failed-rows.md %}) in Soda Cloud.) 
+
+Use one of the following custom metric templates in your [scan YAML]({% link soda-sql/scan-yaml.md %}) file. Replace the values in the double curly braces {%raw %} {{ }} {% endraw %} with your own relevant values.
+
+![template-compatible-1](/assets/images/template-compatible-1.png){:height="170px" width="170px"}
```
Comment (bastienboutonnet) [inline soda-sql/custom-metric-templates.md:37]: I don't think this is great btw as it's not searchable. Is there any reason why we can't inject a text based template instead?

Comment (bastienboutonnet) [reply soda-sql/custom-metric-templates.md:37]: I have a feeling we can probably have some `.md` files live somewhere which we can inject via `{% include file.md %}` no?

File: soda-sql/custom-metric-templates.md [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+        select
+          'found in table 1 but missing in table 2' as directionality,
+          *
+        from table_1_not_in_table_2
+        union all
+        select
+          'found in table 2 but missing in table 1' as directionality,
+          *
+        from table_2_not_in_table_1
+```
```
[the remark this answers is not in the document]
Comment (janet-can) [reply soda-sql/custom-metric-templates.md]: Got it! I separated them.

File: soda-sql/custom-metric-templates.md (hunk 1 above)
Comment (janet-can) [reply soda-sql/custom-metric-templates.md]: I fiddled with what had failed for me last week just adding unicode checkboxes and it's quite a bit closer to the code-tab experience. We can work on making the code-tabs work properly later.

File: soda-sql/custom-metric-templates.md [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+{% endraw %}
+
+<br />
+
+#### Check the sum of column values
+
+<details>
+    <summary>Explain the SQL</summary>
+      The query validates that the sum of two columns in a dataset matches the value in a third column, and identifies those rows which do not match.
+</details>
```
[the remark this answers is not in the document]
Comment (janet-can) [reply soda-sql/custom-metric-templates.md]: Setting up the paradigm on the page to offer explanations in the expand macro is best, I think. With all expanded, it's too wordy; with just a few expanded, it sort of breaks the format expectation. If a reader is curious about how something works, they can quickly access more info. I rather suspect most readers will scan + copy + paste rather than read much, so for them, keeping it tight is ideal.

File: soda-sql/custom-metric-templates.md [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+        left join target_table tt
+            on trim(tt.id) = trim(p.id)
+        where p.id is null
+```
+{% endraw %}
+
+
+## Compare values between datasets
+
+To compare values in columns in different datasets in the same data source, use one of the following custom metric templates in your [scan YAML]({% link soda-sql/scan-yaml.md %}) file. Use one of the following custom metric templates in your [scan YAML]({% link soda-sql/scan-yaml.md %}) file. Replace the values in the double curly braces {%raw %} {{ }} {% endraw %} with your own relevant values.
```
[the remark this answers is not in the document]
Comment (janet-can) [reply soda-sql/custom-metric-templates.md]: Data source is is the warehouse or database.
